File: ios/NotificationPermissionManager.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(NotificationPermissionManager, NSObject)
RCT_EXTERN_METHOD(requestAuthorization:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(scheduleOverrideOrderPrompt:(nonnull NSNumber *)delaySeconds resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(cancelOverrideOrderPrompt:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(cancelAllAppReminders:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
@end

File: ios/undelivery_rn_tmp/NotificationPermissionManager.swift
import Foundation
import UserNotifications
import UIKit
import React

@objc(NotificationPermissionManager)
class NotificationPermissionManager: NSObject {
  @objc(requestAuthorization:rejecter:)
  func requestAuthorization(
    _ resolve: @escaping RCTPromiseResolveBlock,
    rejecter reject: @escaping RCTPromiseRejectBlock
  ) {
    UNUserNotificationCenter.current().requestAuthorization(options: [.alert, .badge, .sound]) { granted, error in
      if let error = error {
        reject("notification_failed", error.localizedDescription, error)
        return
      }
      DispatchQueue.main.async {
        UIApplication.shared.registerForRemoteNotifications()
      }
      resolve(granted)
    }
  }

  @objc static func requiresMainQueueSetup() -> Bool {
    return false
  }
}
